apps/train_loc/lib/train_loc/manager/event.ex

```diff
@@ -6,10 +6,16 @@ defmodule TrainLoc.Manager.Event do
 
   alias TrainLoc.Manager.EventJsonParser
 
+  @type t() :: %__MODULE__{
+          date: String.t() | nil,
+          vehicles_json: [map()] | []
```

Comment:
```suggestion
          vehicles_json: [map()]
```

`[]` is a strict subset of `[map()]` so including it doesn't provide any additional information.